airflow/providers/apache/spark/hooks/spark_sql.py

```diff
@@ -67,27 +67,50 @@ def __init__(
         executor_memory: Optional[str] = None,
         keytab: Optional[str] = None,
         principal: Optional[str] = None,
-        master: str = 'yarn',
```

Comment:
This is not a backwards-compatible change. Can you please make it works also if someone calls it in the old way ? We have some mechanism (just look for "deprecated") that we mark certain parameters as deprecated, but the rule of thumb we have is that we deprecate first and only after next major release we remove the deprecated features. It seems possible to add (and test) the old way of using the hook here.

Also the docstrings were not updated - but they should be - including the deprecation information and explanation what has been replaced.

Reply:
@potiuk reverted to make backwards compatible and add deprecation explanation in docstrings!